controllers/snapshot/snapshot_adapter.go

```diff
@@ -130,6 +137,64 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (results.OperationRes
 	return results.ContinueProcessing()
 }
 
+//EnsureCreationOfEnvironment makes sure that all envrionemnts that were requested via
+//IntegrationTestScenarios get created, in case that environment is already created, provides
+//a message about this fact
+func (a *Adapter) EnsureCreationOfEnvironment() (results.OperationResult, error) {
+
+	integrationTestScenarios, err := helpers.GetAllIntegrationTestScenariosForApplication(a.client, a.context, a.application)
+
+	if err != nil {
+		a.logger.Error(err, "Failed to get Integration test scenarios for following application",
+			"Application.Name", a.application.Name,
+			"Application.Namespace", a.application.Namespace)
+		return results.RequeueOnErrorOrStop(err)
+	}
+
+	allEnvironments, err := a.getAllEnvironments()
+	if err != nil {
+		return results.RequeueOnErrorOrStop(err)
+	}
+
+	if integrationTestScenarios != nil {
+		for _, integrationTestScenario := range *integrationTestScenarios {
+			integrationTestScenario := integrationTestScenario //G601
+			for _, environment := range *allEnvironments {
+				environment := environment //G601
+				//prevent creating already existing environments
+				if helpers.HasLabelWithValue(&environment, "test.appstudio.openshift.io/snapshot", a.snapshot.Name) && helpers.HasLabelWithValue(&environment, "test.appstudio.openshift.io/scenario", integrationTestScenario.Name) {
+					fmt.Println("ENV: " + environment.Name + "Already exists and contains snapshot: " + a.snapshot.Name + " and scenario: " + integrationTestScenario.Name)
```

Comment:
Probably should use logger here.

Reply:
Agreed, we should use the logger for this.

Reply:
done